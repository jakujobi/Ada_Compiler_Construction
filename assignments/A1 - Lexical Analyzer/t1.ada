PROCEDURE Hello IS;
BEGIN
    X := 10;
    Y := 3.5;
    PUT("Hello, Ada!");
END Hello;