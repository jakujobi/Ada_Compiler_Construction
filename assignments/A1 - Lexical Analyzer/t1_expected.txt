Token(type='PROCEDURE', lexeme='PROCEDURE', value=None)
Token(type='ID', lexeme='HELLO', value=None)
Token(type='IS', lexeme='IS', value=None)
Token(type='SYMBOL', lexeme=';', value=None)
Token(type='BEGIN', lexeme='BEGIN', value=None)
Token(type='ID', lexeme='X', value=None)
Token(type='ASSIGN', lexeme=':=', value=None)
Token(type='NUM', lexeme='10', value=10)
Token(type='SYMBOL', lexeme=';', value=None)
Token(type='ID', lexeme='Y', value=None)
Token(type='ASSIGN', lexeme=':=', value=None)
Token(type='REAL', lexeme='3.5', value=3.5)
Token(type='SYMBOL', lexeme=';', value=None)
Token(type='PUT', lexeme='PUT', value=None)
Token(type='SYMBOL', lexeme='(', value=None)
Token(type='LITERAL', lexeme='"Hello, Ada!"', value='Hello, Ada!')
Token(type='SYMBOL', lexeme=')', value=None)
Token(type='SYMBOL', lexeme=';', value=None)
Token(type='END', lexeme='END', value=None)
Token(type='ID', lexeme='HELLO', value=None)
Token(type='SYMBOL', lexeme=';', value=N